### PR TITLE
Use root constraints when they have no intersection with extra.symfony.require

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fixes https://github.com/symfony/flex/issues/712
Fixes https://github.com/composer/composer/issues/9409
Fixes https://twitter.com/scherzmut/status/1323609763125612546

### CODE REVIEW

File: src/Flex.php [hunk 1]
```diff
@@ -863,7 +863,11 @@ public function truncatePackages(PrePoolCreateEvent $event)
         if (!$this->filter) {
             return;
         }
-        $event->setPackages($this->filter->removeLegacyPackages($event->getPackages(), $event->getRequest()->getFixedOrLockedPackages()));
+
+        $rootPackage = $this->composer->getPackage();
+        $lockedPackages = $event->getRequest()->getFixedOrLockedPackages();
```
Comment (Seldaek) [inline src/Flex.php:868]: This is not compatible with composer 1, so you may want to make it optional somehow, Semver\Intervals also does not exist in v1.

I guess it's already broken in the current release tho.. Or is this code path only run for composer 2?

**Comment by Seldaek** [comment]
Looks good to me for v2 that is, I'd say people should upgrade anyway but I am biased.

File: src/Flex.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Flex.php:868]: this codepath runs on c2 only, yes

**Review by jderusse — APPROVED** [review]
I'm fine with fixing this for composer 2 only.

Let people open a PR if they want fixing this issue on composer 1

**Comment by nicolas-grekas** [comment]
OK, support for Composer 1 wasn't that hard, pushed :)

**Pull request merged**